indexer/modules/custom/project_contracts/export_project_contracts_job.py

```diff
@@ -0,0 +1,128 @@
+import logging
+from collections import defaultdict
+from typing import List
+
+from indexer.domain.trace import Trace
+from indexer.domain.transaction import Transaction
+from indexer.executors.batch_work_executor import BatchWorkExecutor
+from indexer.jobs.base_job import ExtensionJob
+from indexer.modules.custom.project_contracts.domain.project_contract_domain import ProjectContractD
+from indexer.modules.custom.project_contracts.models.project_contract import AfProjectContracts
+from indexer.modules.custom.project_contracts.models.projects import AfProjects
+from indexer.utils.abi import bytes_to_hex_str
+
+logger = logging.getLogger(__name__)
+
+
+class ExportProjectContractsJob(ExtensionJob):
+    dependency_types = [Transaction, Trace]
+    output_types = [ProjectContractD]
+    able_to_reorg = False
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self._batch_work_executor = BatchWorkExecutor(
+            kwargs["batch_size"],
+            kwargs["max_workers"],
+            job_name=self.__class__.__name__,
+        )
+
+        self._is_batch = kwargs["batch_size"] > 1
+        self.db_service = kwargs["config"].get("db_service")
+        self.chain_id = self._web3.eth.chain_id
+        self.trace_type_prefix = "create"
+
+        self.project_deployers = defaultdict(list)
+        self.project_contracts = defaultdict(set)
+        self.transaction_map = dict()
+
+        self.read_configured_projects()
+        self.read_discovered_contracts()
+
+    def _collect(self, **kwargs):
+        transactions = self._data_buff[Transaction.type()]
+        transactions = [
+            a_transaction
+            for a_transaction in transactions
+            if a_transaction.receipt and a_transaction.receipt.status == 1
+        ]
+        traces = self._data_buff[Trace.type()]
+
+        self.transaction_map = {tnx.hash: tnx for tnx in transactions}
+        filtered_trace_list = [
+            trace for trace in traces if trace.trace_type.startswith(self.trace_type_prefix) and trace.status == 1
+        ]
+
+        res = []
+        for project_id, deployers in self.project_deployers.items():
+            for deployer in deployers:
+                res.extend(self.direct_create_contracts(project_id, deployer, filtered_trace_list, transactions))
+                res.extend(self.contract_create_contracts(project_id, deployer, filtered_trace_list))
+        self._collect_items(ProjectContractD.type(), res)
```

Comment:
In the end, only keep those with project id?

Reply:
now, only keep those contracts related to project